### PR TITLE
Differentiating the artifacts deployed with CAR file and not persist the changes into the default location.

### PR DESCRIPTION
**Pull request by kishokpriya**

### CODE REVIEW

**Comment by chanakaudaya** [comment]
@kishokpriya Please remove the formatting changes in this commit. There are a lot of formatting changes without any effect to the fix.

File: components/mediation-initializer/org.wso2.carbon.mediation.initializer/src/main/java/org/wso2/carbon/mediation/initializer/utils/CAppArtifactsMap.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class CAppArtifactsMap {
+
+    /* Map to hold CApp artifact details for specific tenants*/
+    Map<String, CAppArtifactData> cAppArtifactDataMap = new HashMap<String, CAppArtifactData>();
+
+    public Map<String, CAppArtifactData> getcAppArtifactDataMap() {
+        return cAppArtifactDataMap;
+    }
+
+    public void setcAppArtifactDataMap(String key, CAppArtifactData artifactData) {
```
Comment (isudana) [inline components/mediation-initializer/org.wso2.carbon.mediation.initializer/src/main/java/org/wso2/carbon/mediation/initializer/utils/CAppArtifactsMap.java]: This method doesn't set artifact Map rather it add artifactData obj against a key.
method should be renamed to addCAppArtifactData()

File: components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java [hunk 2]
```diff
@@ -134,6 +145,8 @@ public void deployArtifacts(CarbonApplication carbonApp, AxisConfiguration axisC
                 }
             }
         }
+        // Add the CApp artifacts details for specific tenant
+        cAppArtifactDataService.addCAppArtifactData(tenantId, cAppArtifacts);
```
Comment (isudana) [inline components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java]: This means only a tenant can have 1 CAPP ?

File: components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/internal/DataHolder.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package org.wso2.carbon.application.deployer.synapse.internal;
```
Comment (isudana) [inline components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/internal/DataHolder.java]: Shall we put class comments

File: components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java [hunk 4]
```diff
@@ -55,6 +55,9 @@
 import org.wso2.carbon.mediation.initializer.ServiceBusUtils;
```
Comment (isudana) [inline components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java]: Correct formatting issues in all the classes. Some lines are exceeding the allowed limit.

File: components/mediation-initializer/org.wso2.carbon.mediation.initializer/src/main/java/org/wso2/carbon/mediation/initializer/utils/CAppArtifactsMap.java (hunk 1 above)
Comment (kishokpriya) [inline components/mediation-initializer/org.wso2.carbon.mediation.initializer/src/main/java/org/wso2/carbon/mediation/initializer/utils/CAppArtifactsMap.java]: done

File: components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java (hunk 2 above)
Comment (kishokpriya) [inline components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java]: Keeping 1 map for each tenant & holding all CAPP artifacts inside the map. 
When ever the CAPP is uploaded, the artifacts will be append in the existing map.
The implementation is added in 'CAppDataHolder' class under 'addCAppArtifactData' method.

File: components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java (hunk 4 above)
Comment (kishokpriya) [inline components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java]: Done

File: components/mediation-admin/org.wso2.carbon.task/src/main/java/org/wso2/carbon/task/TaskManager.java [hunk 5]
```diff
@@ -109,19 +118,30 @@ public void deleteTaskDescription(String name, String group) {
...
+        CAppArtifactDataService cAppArtifactDataService = ConfigHolder.getInstance().
+                getcAppArtifactDataService();
+        int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+        String artifactName = getArtifactName(artifactType, taskDescription.getName());
         if (className != null && !"".equals(className)) {
             taskManagementServiceHandler.editTaskDescription(taskDescription, className);
+            if (cAppArtifactDataService.isArtifactDeployedFromCApp(tenantId, artifactName)) {
+                cAppArtifactDataService.setEdited(tenantId, artifactName);
+                MediationPersistenceManager pm = ServiceBusUtils.getMediationPersistenceManager(axisCfg);
+                pm.deleteItem(taskDescription.getName(), taskDescription.getName() + ".xml",
```
Comment (nadeeshaan) [inline components/mediation-admin/org.wso2.carbon.task/src/main/java/org/wso2/carbon/task/TaskManager.java]: Can you please explain this logic of deleting the task description xml?

**Comment by VijithaEkanayake** [comment]
Please resolve the conflicts and reopen the PR

**Pull request closed without merging**